wallet/src/main/java/com/android/identity_credential/wallet/ui/prompt/biometric/BiometricPrompt.kt

```diff
@@ -8,13 +8,59 @@ import androidx.biometric.BiometricPrompt.PromptInfo
 import androidx.core.content.ContextCompat
 import androidx.fragment.app.FragmentActivity
 import com.android.identity.android.securearea.UserAuthenticationType
+import com.android.identity_credential.wallet.R
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+/**
+ * Show the Biometric prompt
+ *
+ * Async extension function of a [FragmentActivity] to show Biometric Prompt and return the result as a [Boolean] of whether
+ * authentication was successful, or raises/throws an Exception, such as [IllegalStateException],
+ * if an error prevented the Biometric Prompt from showing.
+ *
+ * @param title the title for the authentication prompt.
+ * @param subtitle the subtitle for the authentication prompt.
+ * @param cryptoObject a crypto object to be associated with this authentication.
+ * @param userAuthenticationTypes the set of allowed user authentication types, must contain at
+ *                                least one element.
+ * @param requireConfirmation option to require explicit user confirmation after a passive biometric.
+ * @return a [Boolean] indicating whether biometric authentication was successful.
+ * @throws Exception if there were errors showing the prompt
+ */
+suspend fun FragmentActivity.showBiometricPrompt(
```

Comment:
As per [this comment](https://github.com/openwallet-foundation-labs/identity-credential/pull/631#discussion_r1629662358) please don't make this an extension function, just make it take the `FragmentActivity` as the first parameter.